scripts/velocity.py

```diff
@@ -0,0 +1,446 @@
+    print(f'''
+Of the {len(prs)} pull requests closed in this reporting period, {len(closed)}
+pull requests were closed but not merged. Points for these PRs are not
+counted towards contributor totals above.
+''')
+    print(print_prs(*closed))
+
+    print(f'''## Resolved issues
+At the time this report was generated, {len(issues)} were marked closed in \
+the reporting period. This indicates that the associated work is considered \
+to have "landed" and been demoed.
+
+| Assignees | Issue | Estimate |
+| --------- | ----- | :------: |''')
+    for issue in issues:
+        authors = [f'[@{a.login}]({a.html_url})' for a in issue.assignees]
+        cols = [', '.join(authors), issue.markdown(), str(issue.estimate or '')]
+        print('', *cols, '', sep='|')
+
+
+def configure():
+    print('To retrieve issue estimates, you need a ZenHub API key. You can '
+          'generate one here: https://app.zenhub.com/dashboard/tokens')
+    config.auth['zenhub_key'] = getpass.getpass('ZenHub API key (hidden): ')
+    print('To retrieve PR-issue relationships, you need a ZenHub key for their '
+          'private API. In your browser, open ZenHub, then open the developer '
+          'tools pane and inspect network requests. Click an issue, and look for '
+          'a request to `events`. The key we want is in the  '
+          'X-Authentication-Token header.')
+    config.auth['zenhub_key_private'] = getpass.getpass('ZH private API key: ')
+    ws = config.parse_workspace_url(input('ZenHub workspace URL (all repos selected): '))
+    config.auth['github_user'] = input('GitHub username: ')
+    print('To retrieve issue data, you need a GitHub personal access key. The key '
+          'requires the `public_repo` scope at minimum, or the entire `repo` scope '
+          'to access private repository data. You can generate an access key here: '
+          'https://github.com/settings/tokens')
+    config.auth['github_key'] = getpass.getpass('GitHub personal access key (hidden): ')
+    config.workspace[ws['name']] = ws
+    config.save()
+    print(f"\nConfigured with workspace {ws['name']!r}, connected to "
+          f"{len(ws['repos'])} repositories. Configuration saved to {config.path}.")
+
+
+@functools.lru_cache
+def get_user_by_id(user_id: int) -> str:
+    # ZenHub identifies users only by their GitHub user ID. The only way to map
+    # a user ID to a user name is to use an undocumented API endpoint that
+    # PyGithub does not directly support. (PyGithub/PyGithub#1615)
+    r = requests.get(f'https://api.github.com/user/{user_id}',
+                     auth=(config.auth['github_user'], config.auth['github_key']))
+    r.raise_for_status()
+    return r.json()['login']
+
+
+def _human_readable_date(since: datetime.datetime,
+                         until: datetime.datetime) -> str:
+    """
+    >>> _human_readable_date(datetime.datetime.now() - datetime.timedelta(days=25),\
+                             datetime.datetime.now())
```

Comment:
```suggestion
def _human_readable_date(since: datetime.datetime, until: datetime.datetime) -> str:
    """
    >>> _human_readable_date(datetime.datetime.now() - datetime.timedelta(days=25), datetime.datetime.now())
```

Reply:
Suggestion retracted.